notebooks/stitches_dev/Notebook6_throwout_Duplicates-across-ensemble-members.Rmd

```diff
@@ -0,0 +1,297 @@
+
+But this does then take us from having a huge number of all possible mixes and matches in terms of generated realizations to only being able to generate a much smaller number of ensemble members: the target year with the smallest number of matching archive points dictates the number of realizations that you can generate according to throwing out each (target-year, archive-match) pair once they've been used. So going from like 10^42 possible mixes and matches to just 10 because target year 1989 only had 10 archive points matched in the neighborhood. Every time you do that sampling, those 10 might be different. But you can't have a generated ensemble of more than 10 realizations if each realization has to have a different 1989 value to avoid envelope collapse.
+
+We _can_ define a re-matching condition on a subsetted archive (with existing matches removed) the same way `remove_duplicates` does if we want more than those 10; the issue there is that by definition, those new matches are less good and you are more likely to create unacceptable generated realizations. So for now, not doing it. 
+
+## across targets
+
+Say we want to generate multiple new ensemble members for each of a target trajectory of realization 1 and a target trajectory of realization 4. Because those targets are pretty similar, we risk envelope collapse again. THinking of just one generated ensemble member for each target, we don't want the case of having (target realization 1 year 2070) getting the same archive match as (target realization 4 year 2070), in addition to the fact that we don't want that to happen to multiple generated realizations for any of the targets. It does appear to have the same solution then - remove each (target-year, archive-match) pair as they get used, regardless of which trajectory the target-year came from.
+
+So, we construct generated realizations iteratively. We start with the target that has the most limited number of collapse-free generated ensemble members. Construct those, removing each constructed (target-year, archive-match) as we go. Then move on to the next target with the next smallest number of collapse-free generated ensemble members. Repeat until we're either done or out of collapse-free ensembles to construct. I could see a scenario where we run out of collapse-free ensembles before we run out of targets to match, just depending on archive/sampling/etc/.
+
+Note that because we want to enforce this across targets, it's really not parallellizable except across _experiments_. Also in general, with the way the code has been written, it's probably best to call it separately for each experiment of interest being worked with. Just because this whole duplication elimination stuff is not optional and it wouldn't make sense that points used in the construction of a new ssp245 realization couldn't be used in the construction of a new ssp585. 
+
```

Comment:
this is just saying that it only eliminates duplicates for a given target time series, right? but we are always going to have a single time series (target scenario) being read in at a time right?